esp/bindings/http/platform/httptransport.cpp

```diff
@@ -1915,11 +1915,16 @@ int CHttpRequest::receive(IMultiException *me)
     return 0;
 }
 
-void CHttpRequest::startSpan()
+void CHttpRequest::startSpan(const char * serviceName, const char * methodName)
 {
     //MORE: The previous code would be better off querying httpHeaders...
+    StringBuffer spanName;
+    spanName.append(serviceName);
+    if (!isEmptyString(methodName))
```

Comment:
@afishbeck depending on how the service function is called, sometimes the methodName is not known at this point (part of the soap envelope?)  How can it be cleanly extracted?  May require some refactoring, so a different PR.

Reply:
Yes this timing issue related to when we know the name comes up sometimes.  You could open a JIRA to refactor to always get the name at the beginning.  I actually did similar refactoring in roxie.

One idea would be to do a quick pull parser extraction of the name even if it came from the SOAP/XML/JSON content of the message body.  Generally the pull parser would usually only have to walk a few tags to find what it needs.  It could then abandon the parsing until the body is fully parsed later.

Another way would be to always parse the body much sooner.  Early on we decided to avoid extra overhead in case we rejected the request for other reasons.. for example security.  We save a little time and memory, or perhaps there may occasionally be huge requests.  I think Roxie parses pretty early but isn't as close to the front lines as ESP.

Either way would be good in the sense that it would let us refactor some other code that has to wait until the method is known.

Reply:
See QueryNameExtractor in roxie.

Reply:
HPCC-31013 created.